### PR TITLE
Finish app

### PR DESCRIPTION
**Pull request by helljack26**
Hi Vlad
I have finished this app.
List of added item from previous pull request
1. Query to api by input.
2. Made one request to api, from two similar
3. Added condition to render genres, if it haven't, it doesn't render.
4. Corrected header component render .
5. Added page header to state. Now on Popular page and Search page render correct header.

Since last pull request I have added
1. localStorage and watch list logic.
2. Media screen for all screens till 300px.
3. Tooltip for add to watch list button.

Also there have couple mistakes and trouble that I saw and havent change
1. If on film detail page you will reload page, you will see infinite loading. It can resolve if saving this page in localStorage
and render if page have reload.
2. Also function that get watch list from localStorage, was multiple cloned in three places and wasn't optimized.
3. Tooltip for button add to watch list, when you click on button, tooltip have shifted because content in buttons had different size.
4. Also I havent add file with name 'consts' for all const that I have in project.

### CODE REVIEW

File: src/components/AddToWatchListButton/AddToWatchListButton.jsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
+
+import { useState } from 'react'
+import { useDispatch } from 'react-redux';
+import style from './AddToWatchListButton.module.css'
+import { toWatchList } from '../../store/filmApi/types';
+
+const buttonTypeArr = {
```
Comment (Oodmincheg) [inline src/components/AddToWatchListButton/AddToWatchListButton.jsx]: Helloo ) Why `Arr` if it's object. And it's necessary to add type of var to its name :)

File: src/components/AddToWatchListButton/AddToWatchListButton.jsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        text: <span>Add to<br /> watch list</span>,
+        class: 'notInWatch'
+    }
+}
+
+const AddToWatchListButton = ({ name, image, genre, id, buttonType }) => {
+    const dispatch = useDispatch();
+    const [type, setType] = useState();
+    const [tooltipText, setTooltipText] = useState(buttonTypeArr.notInWatch.tooltip);
+    switch (type === undefined ? buttonType : type) {
```
Comment (Oodmincheg) [inline src/components/AddToWatchListButton/AddToWatchListButton.jsx]: Very confusing using of switch. To be honest I can't understand anything :D Are you sure that there is not clearer way to do the job?)

File: src/components/AddToWatchListButton/AddToWatchListButton.jsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        case 'inWatch':
+        case true:
+            return (
+                <>
+                    <button type='button' className={style.addIconBtn}>
+                        <img src={buttonTypeArr.inWatch.url} alt={buttonTypeArr.notInWatch.tooltip}
```
Comment (Oodmincheg) [inline src/components/AddToWatchListButton/AddToWatchListButton.jsx]: At least move this buttons to the separate components with nice naming :)

File: src/components/AddToWatchListButton/AddToWatchListButton.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                            className={style.inWatch}
+                            onClick={() => {
+                                dispatch(toWatchList(undefined, undefined, undefined, id))
```
Comment (Oodmincheg) [inline src/components/AddToWatchListButton/AddToWatchListButton.jsx]: triple undefined O_O. We definitely should discuss this part on the lesson.

File: src/components/FilmDetailsPage/FilmDetailsPage.jsx [hunk 5]
```diff
@@ -1,69 +1,55 @@
...
-            setDetails( data );
-        };
-        getFilmDetails();
-    }, [id] );
 
+const FilmDetailsPage = () => {
+    const filmDetails = useSelector(state => state.filmApi.filmDetails)
+    const loading = useSelector(state => state.filmApi.loading)
+    function checkInLocalStorage(id) {
+        const localStorageWatchList = window.localStorage.getItem('watchList')
```
Comment (Oodmincheg) [inline src/components/FilmDetailsPage/FilmDetailsPage.jsx:21]: First variable is JSON and second already parsed JSON :)

File: src/components/FilmDetailsPage/FilmDetailsPage.jsx [hunk 6]
```diff
@@ -1,69 +1,55 @@
...
+        const localStorageWatchListJson = JSON.parse(localStorageWatchList);
+        const filmFromLocalStorage = localStorageWatchListJson !== null ? localStorageWatchListJson.find(film => film.id === id) : false
+        const buttonType = Boolean(filmFromLocalStorage) === true ? 'detailsInWatch' : 'detailsNotInWatch'
+        return buttonType
+    }
+    // For render page 
+    const genreArr = filmDetails.genres !== undefined ? translateGenre(filmDetails.genres.map((item) => item.id)) : null
```
Comment (Oodmincheg) [inline src/components/FilmDetailsPage/FilmDetailsPage.jsx:29]: what if filmDetails will be null or string?

File: src/components/FilmListCard/FilmListCard.jsx [hunk 7]
```diff
@@ -1,29 +1,25 @@
...
-    // If havent poster return nothing
+    const genreArr = genre.length !== 0 ? <p>Genre: {translateGenre(genre)}</p> : null;
     return image !== null ? (
         <>
-            <div className={style.cardBlock} onClick={() => dispatch( updatePageTitle( 'card' ) )}>
-                <Link to={`/${id}`} className={style.filmListCard}>
-                    {/* Poster */}
+            <div className={style.cardBlock}>
+                <Link to={`/${id}`} className={style.filmListCard}
+                    onClick={() => dispatch(getFilmDetailsFromApi(id))}>
```
Comment (Oodmincheg) [inline src/components/FilmListCard/FilmListCard.jsx:15]: I'm not sure, but it seems that this should be handle by router param.

File: src/components/AddToWatchListButton/AddToWatchListButton.jsx (hunk 2 above)
Comment (helljack26) [reply src/components/AddToWatchListButton/AddToWatchListButton.jsx]: I have split one component to two different component for card and for details.

File: src/components/FilmDetailsPage/FilmDetailsPage.jsx (hunk 5 above)
Comment (helljack26) [reply src/components/FilmDetailsPage/FilmDetailsPage.jsx:21]: Resolve it

File: src/components/FilmDetailsPage/FilmDetailsPage.jsx (hunk 6 above)
Comment (helljack26) [reply src/components/FilmDetailsPage/FilmDetailsPage.jsx:29]: add confition if null and if string

**Pull request merged**